client.js

```diff
@@ -0,0 +1,145 @@
+var scene = new THREE.Scene();
+var camera = new THREE.PerspectiveCamera( 75, window.innerWidth / window.innerHeight, 0.1, 1000 );
+
+var renderer = new THREE.WebGLRenderer();
+renderer.setSize( window.innerWidth, window.innerHeight );
+document.body.appendChild( renderer.domElement );
+
+// arrow rendering parameters
+var maxArrowLength = 0.375;
+var hex = 0xffff00;
+var maxHeadLength = 0.5 * maxArrowLength;
+var maxHeadWidth = 0.5 * maxArrowLength;
+
+camera.position.z = 10;
+
+class Plotter {
+    constructor() {
+        this.vectorFieldSamples = [];
+
+        this.chargeGeometry = new THREE.CircleGeometry( 0.1, 32 );
+        this.chargeMaterial = new THREE.MeshBasicMaterial( { color: 0x0000FF} );
+        this.pointCharge = new THREE.Mesh( this.chargeGeometry, this.chargeMaterial );
+        scene.add( this.pointCharge );
+    }
+
+    updatePlot(newVectorFieldSamples, newChargePosition) {
+        for (var arrow of this.vectorFieldSamples) {
+            scene.remove(arrow);
+        };
+
+        this.vectorFieldSamples = [];
+
+        for (var newSample of newVectorFieldSamples) {
```

Comment:
The innards of this for-loop does 1 thing. Could extract into a function: `addArrowToSceneAndSampleList`